test/pub_update_test.dart

```diff
@@ -129,6 +130,56 @@ void main() {
         );
       });
 
+      test(
+        '''returns true when currentVersion is pre release and latestVersion is stable''',
+        () async {
+          expect(
+            await pubUpdater.isUpToDate(
+              packageName: 'very_good_cli',
+              currentVersion: '0.4.0-dev.1',
+            ),
+            true,
+          );
+        },
+      );
+
+      group('pre releases', () {
+        setUp(() {
+          client = MockClient();
+          response = MockResponse();
+          pubUpdater = PubUpdater(client);
+          processManager = MockProcessManager();
```

Comment:
I would love to have these mocks be private.

Reply:
Awesome, maybe on another pr, so the diff here is minimal.